### PR TITLE
DAOS-9088 docker: use rocky8.4 as default image

### PR DESCRIPTION
**Pull request by knard-intel**
Change the default base image used by the virtual docker cluster: switch
from centos8 to rocky8.4. Use authentification as default choice.

The modification were validated with centos7 and centos8 baremetal nodes (i.e. wolf-67 and wolf-70)

doc-only: true

Signed-off-by: Cedric Koch-Hofer <cedric.koch-hofer@intel.com>

### CODE REVIEW

**Review by johannlombardi — APPROVED** [review]

File: utils/docker/vcluster/README.md [hunk 1]
```diff
@@ -24,18 +24,19 @@ At this time only emulated hardware storage are supported by this Docker platfor
 
 To build and deploy the Docker images, `docker` and optionally `docker-compose` shall be available.
 The docker host should have access to the [Docker Hub](https://hub.docker.com/) and
-[CentOS](https://www.centos.org/) official repositories.  Finally,
+[Rocky Linux](https://rockylinux.org/fr/) official repositories.  Finally,
```
Comment (ashleypittman) [inline utils/docker/vcluster/README.md:27]: We should probably link to the english version of the webstite.
```suggestion
[Rocky Linux](https://rockylinux.org/) official repositories.  Finally,
```

File: utils/docker/vcluster/README.md [hunk 2]
```diff
@@ -24,18 +24,19 @@ At this time only emulated hardware storage are supported by this Docker platfor
...
+The platform was tested and validated with the
+[rockylinux/rockylinux:8.4](https://hub.docker.com/r/rockylinux/rockylinux) and
+[centos:centos8](https://hub.docker.com/_/centos) official docker images.  However other RHEL-like
+distributions should be supported.
 
 !!! warning
     Some distributions are not yet well supported such as
-    [rocky8.5](https://hub.docker.com/r/rockylinux/rockylinux): issue with the management of
-    hugepages with the [spdk](https://spdk.io/) library.
+    [rockylinux/rockylinux:8.5](https://hub.docker.com/r/rockylinux/rockylinux): issue with the
```
Comment (ashleypittman) [inline utils/docker/vcluster/README.md:38]: Almost all of this PR refers to 8.4, this line however says 8.5

**Review by ashleypittman — COMMENTED** [review]
It might be better to just upgrade everything to 8.5 if we plan to stick to Rocky going forwards.

**Comment by knard-intel** [comment]
> It might be better to just upgrade everything to 8.5 if we plan to stick to Rocky going forwards.

At this time, we are facing some issues with rocky8.5 (as explained in the README): issue with spdk and huge pages.
However, rocky8.5 is working well on baremetal and thus this issue will be solved in another US.

File: utils/docker/vcluster/README.md (hunk 2 above)
Comment (knard-intel) [reply utils/docker/vcluster/README.md:38]: As indicated in my previous comment, this sentence sadly indicate that we are not yet supporting rocky8.5.

File: utils/docker/vcluster/README.md (hunk 1 above)
Comment (knard-intel) [reply utils/docker/vcluster/README.md]: fixed with commit https://github.com/daos-stack/daos/pull/7640/commits/04f06f6804d065f368b0138af0d3dce0462a4484

File: utils/docker/vcluster/README.md (hunk 2 above)
Comment (ashleypittman) [reply utils/docker/vcluster/README.md:38]: Ah, so the issue here is not with Rocky it's with the 8.5 version of rocky.  I'd misunderstood this comment.

**Review by ashleypittman — APPROVED** [review]

File: utils/docker/vcluster/README.md (hunk 2 above)
Comment (johannlombardi) [reply utils/docker/vcluster/README.md:38]: yeah, baby steps, but we will get there.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**